x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/Values.java

```diff
@@ -115,4 +116,9 @@ public AggregatorFunctionSupplier supplier(List<Integer> inputChannels) {
         // TODO cartesian_point, geo_point
         throw EsqlIllegalArgumentException.illegalDataType(type);
     }
+
+    @Override
+    public Expression surrogate() {
+        return field().foldable() ? field() : null;
```

Comment:
Values not only merges values, but also removes duplicates (If no test was triggered because of this, we should add some!)

```
ROW x = [1, 1, 2] | STATS a = VALUES(x)

-> [1, 2]
```

Reply:
Good catch, BUT I think we have a problem with the documentation. It's not mentioning this aspect. There were other misses in our functions docs (which are fixed in this PR), I think we need to review our documentation on functions and double check its correctness and completeness. I will create an issue.

Reply:
@ivancea thank you for pro-actively checking this PR 🙏, that was very helpful.
I've created two issues:
- improving our documentation: https://github.com/elastic/elasticsearch/issues/112437
- this PR also now depends on pending `mv_values` addition: https://github.com/elastic/elasticsearch/issues/112445